### PR TITLE
Replace a lot of any with VgSignal, add log message to log.ts

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compile/compile.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ function assembleTopLevelModel(model: Model, topLevelProperties: TopLevelPropert
   // move width and height signals with values to top level
   layoutSignals = layoutSignals.filter(signal => {
     if ((signal.name === 'width' || signal.name === 'height') && signal.value !== undefined) {
-      topLevelProperties[signal.name] = signal.value;
+      topLevelProperties[signal.name] = +signal.value;
```
Comment (kanitw) [inline src/compile/compile.ts]: Shoudn't signal values already be a number?

File: src/vega.schema.ts [hunk 2]
```diff
@@ -59,32 +59,32 @@ export function isVgSignalRef(o: any): o is VgSignalRef  {
...
+  sort?: VgUnionSortField;
+}
 
 // TODO: add vg prefix
-export type FieldRefUnionDomain = {
-  data: string,
-  fields: VgFieldRef[],
-  sort?: VgUnionSortField
-};
+export interface FieldRefUnionDomain {
```
Comment (kanitw) [inline src/vega.schema.ts]: Just do the Vg prefix now ?

File: src/compile/compile.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/compile.ts]: Can't they be strings or bools?

Comment (kanitw) [reply src/compile/compile.ts]: By type, yes.  By how we initialize width and height signals, should be no (unless we have a bug).

File: src/vega.schema.ts (hunk 2 above)
Comment (domoritz) [reply src/vega.schema.ts]: Done

File: src/compile/compile.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/compile.ts]: Right, this is just a type thing. I guess we could say `as number`

**Pull request merged**